ceno_emul/src/platform.rs

```diff
@@ -60,7 +66,7 @@ impl Platform {
     // Permissions.
 
     pub fn can_read(&self, addr: Addr) -> bool {
-        self.is_rom(addr) || self.is_ram(addr) || self.is_pub_io(addr)
+        self.is_rom(addr) || self.is_ram(addr) || self.is_pub_io(addr) || self.is_priv_io(addr)
```

Comment:
We might want to make the emulator match what the circuits allow?  Mostly to simplify reasoning about soundness.

As far as I can tell from what you wrote, only some parts of the `private_io` address range are actually readable?  (And I suspect it's similar for the other address ranges as well?)

Reply:
That would be ideal yes.

Reply:
Tracked in https://github.com/scroll-tech/ceno/issues/630